src/Selenium2Driver.php

```diff
@@ -131,7 +131,7 @@ public function setDesiredCapabilities($desiredCapabilities = null)
         // See https://sites.google.com/a/chromium.org/chromedriver/capabilities
         if (isset($desiredCapabilities['chrome'])) {
 
-            $chromeOptions = array();
+            $chromeOptions = (isset($desiredCapabilities['chromeOptions']) && is_array($desiredCapabilities['chromeOptions']))? $desiredCapabilities['chromeOptions']:array();
 
             foreach ($desiredCapabilities['chrome'] as $capability => $value) {
```

Comment:
The `w3c` option should go into `$desiredCapabilities['chrome']` array to work correctly.

Are you sure Behat (through MinkExtension) have no way of populating that array?

Here is relevant code fragment, that builds that array for the driver on Behat side:

https://github.com/Behat/MinkExtension/blob/master/src/Behat/MinkExtension/ServiceContainer/Driver/Selenium2Factory.php#L51-L86

Reply:
The code fragment, show how the content of the array `extra_capabilities` defined into the Behat configuration will be add into the array fo `$desiredCapabilities` variable.

But the line `$chromeOptions = array();` initilize an new array without check if the current configuration have already this array.
And the line `$desiredCapabilities['chromeOptions'] = $chromeOptions;` set an new value without merge the content.

Reply:
@aik099 are my answers satisfactory?